### PR TITLE
Mix task for tests, mocks and integration

### PR DESCRIPTION
**Pull request by arjun289**
Pull request covers a mix task to generate the following files for the gateway:
1. Test
2. Integration Test
3. Mock Responses.

### CODE REVIEW

File: templates/integration.eex [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  describe "capture" do 
+  end
+
+  describe "void" do
+  end
+
+  describe "refund" do
+  end
+
+  describe "environment" do
```
Comment (oyeb) [inline templates/integration.eex]: What's environment?

File: templates/integration.eex [hunk 2]
```diff
@@ -0,0 +1,39 @@
+defmodule Gringotts.Integration.Gateways.<%= gateway_module <> "Test"%> do
+  # The following module should include the integration tests
+  # for the <%= gateway_module%> 
+  # It contains boilerplate which helps with writing the tests.
```
Comment (oyeb) [inline templates/integration.eex]: Reword:
> Integration tests for <%= gateway_module %>.

Remove "It contains ..."

File: templates/mock_response.eex [hunk 3]
```diff
@@ -0,0 +1,9 @@
+defmodule Gringotts.Gateways.<%= gateway_module <> "Mock"%> do
+
+  # The module should include mock responses for test cases in <%= gateway_mock_test <> ".exs"%>.
+  # e.g. 
+  # def successful_purchase do 
+  #  {:ok, %HTTPoison.Response{body: "{data: "successful_purchase"}"}
```
Comment (oyeb) [inline templates/mock_response.eex]: Escape the quotes,
```elixir
%HTTPoison.Response{body: ~s[{data: "successful_purchase"}]}
```

File: templates/test.eex [hunk 4]
```diff
@@ -0,0 +1,29 @@
+defmodule Gringotts.Gateways.<%= gateway_module <> "Test" %> do
+  # The following file should include tests for the <%= gateway_module%>
```
Comment (oyeb) [inline templates/test.eex]: Reword:
```
# This file contains mocked tests for <%= gateway_module %>

# We recommend using [mock][1] for this, you can place the mock responses from
# the Gateway in `test/mocks/<%= gateway_mock_response%>.exs` file, which has also been
# generated for you.
#
# [1]: https://github.com/jjh42/mock
```

File: templates/integration.eex (hunk 1 above)
Comment (arjun289) [reply templates/integration.eex]: renamed this to environment setup

**Pull request merged**